docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,231 @@
+
+:::code language="csharp" source="snippets/whats-new-csharp9/RecordsExamples.cs" ID="RecordDefinition":::
+
+That record definition creates a `Person` type that contains two readonly properties: `FirstName` and `LastName`. The `Person` type is a reference type. If you looked at the IL, it’s a class. It’s immutable in that none of the properties can be modified once it's been created. When you define a record type, the compiler synthesizes several other methods for you:
+
+- Methods for value-based equality comparisons
+- Override for GetHashCode
+- Copy and Clone members
+- PrintMembers and ToString
+- Deconstruct method
+
+Records support inheritance. You can declare a new record derived from `Person` as follows:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/RecordsExamples.cs" ID="InheritedRecord":::
+
+You can also seal records to prevent further derivation:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/RecordsExamples.cs" ID="SealedRecord":::
+
+The compiler synthesizes different versions of the methods above. The method signatures depend on if the record type is sealed and if the direct base class is object. Records should have the following capabilities:
+
+- Equality is value-based, and includes a check that the types match. For example, a `Student` can't be equal to a `Person`, even if the two records share the same name.
+- Records have a consistent string representation generated for you.
+- Records support copy construction. Correct copy construction must include inheritance hierarchies, and properties added by developers.
+- Records can be copied with modification. These copy and modify operations supports non-destructive mutation.
+- All records support deconstruction.
+
+In addition to the familiar `Equals` overloads, `operator ==` and `operator !=`, the compiler synthesizes a new `EqualityContract` property. The property returns a `Type` object that matches the type of the record. If the base type is `object`, the property is `virtual`. If the base type is another record type, the property is an `override`. If the record type is `sealed`, the property is `sealed`. The synthesized `GetHashCode` uses the `GetHashCode` from all the public properties declared in the base type and the record type. These synthesized methods enforce value-based equality throughout an inheritance hierarchy. That means a `Student` will never be considered equal to a `Person` with the same name. The types of the two records must match as well as all properties shared among the record types being equal.
```

Comment:
I believe GetHashCode accounts for the synthesized EqualityContract property as well, not only the properties declared by the developer.

Reply:
Note: the text says GetHashCode uses the public properties.
But I used a private property in SharpLab and it seems GetHashCode uses this private property. Will that change in the official release? If not, the text should be updated.